packages/styled-components/src/types.ts

```diff
@@ -285,3 +285,12 @@ export type CSSProp = Interpolation<any>;
 export type NoInfer<T> = [T][T extends any ? 0 : never];
 
 export type Substitute<A extends object, B extends object> = FastOmit<A, keyof B> & B;
+
+type AnyIfEmpty<T extends object> = keyof T extends never ? any : T;
+
+export interface ThemeProps<T> {
```

Comment:
This could definitely be an alias to something else like `ExecutionContext`

Reply:
But it'd need on top of #4103 so we can have the first generic param to `ExecutionContext` be the Theme

Reply:
Should it be defined as an alias? Or should I add to the migration guide?

Reply:
Is it possible to change the direction of the branch?

Reply:
@probablyup 
#4103 is still WIP, so I felt it wouldn't be too late to merge this PR first. 
Until then, how about exporting the conventional `ThemeProps` once?

What do you think??

Reply:
When I upgraded from v5 to v6, I replaced `ThemeProps<T>` with `ExecutionContext` in my codebase. 

If we do decide to add this to ease upgrading that it should be aliased instead of declaring a new type. For example:

```ts
export interface ThemeProps extends ExecutionContext {}
```

Reply:
I've done the same in my codebase, but I'm afraid it won't accept generics.
I think I'll either wait for the ExecutionContext to accept generics in #4103, or temporarily include the code in its current state and include it later in #4103.